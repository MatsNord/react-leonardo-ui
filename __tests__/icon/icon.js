import React from 'react';
import renderer from 'react-test-renderer';

import Icon from '../../src/icon';

test('Normal icon renders correctly', () => {
  const tree = renderer.create(
    <Icon name="close" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Small icon renders correctly', () => {
  const tree = renderer.create(
    <Icon name="close" size="small" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Large icon renders correctly', () => {
  const tree = renderer.create(
    <Icon name="close" size="large" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Missing icon renders correctly', () => {
  const tree = renderer.create(
    <Icon />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
