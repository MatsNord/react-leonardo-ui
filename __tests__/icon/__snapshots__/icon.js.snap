exports[`test Large icon renders correctly 1`] = `
<span
  aria-hidden="true"
  className="lui-icon lui-icon--close lui-icon--large" />
`;

exports[`test Missing icon renders correctly 1`] = `
<span
  aria-hidden="true"
  className="lui-icon" />
`;

exports[`test Normal icon renders correctly 1`] = `
<span
  aria-hidden="true"
  className="lui-icon lui-icon--close" />
`;

exports[`test Small icon renders correctly 1`] = `
<span
  aria-hidden="true"
  className="lui-icon lui-icon--close lui-icon--small" />
`;
